src/Build/Definition/Toolset.cs

```diff
@@ -1062,7 +1062,7 @@ private void LoadAndRegisterFromTasksFile(string[] defaultTaskFiles, ILoggingSer
                             preserveFormatting: false);
                     }
 
-                    foreach (ProjectElement elementXml in projectRootElement.Children)
+                    foreach (ProjectElement elementXml in projectRootElement.ChildrenInternal)
```

Comment:
I guess the only benefits you're getting is here. But I'm curious if you can avoid the collection allocation here as well by having a property 'ChildrenEnumerator' that will return your struct enumerator?

Reply:
Renamed `ChildrenInternal` -> `ChildrenEnumerable` for clarity